### DISCUSSION TITLE
Is there a way to filter databases?

### DISCUSSION THREAD

**Discussion by WorstPIlot** [zubkov-andrei/pg_profile#78] [Q&A] [answered]
Is there a way to get report for one database, instead of full list of databases on server?

**Comment by zubkov-andrei** [comment]
Hi!
Right now this is impossible. The main reason here is statistics consistency. Many statistics in postgres are collected on cluster-level. Once you limit the report or the collection process by some databases, you can't use cluster statistics, because the will show workload of the whole cluster. This can cause misunderstanding of a report.

**Reply by WorstPIlot** [reply]
Is it possible to implement this functionality in the future? Maybe with some warning placed in resulting report?

**Reply by zubkov-andrei** [reply]
Can you describe your use case little bit detailed? Why do you need to limit the databases shown in report?

**Reply by WorstPIlot** [reply]
We use this software to provide clients with information about the load and status of databases, but unfortunately, since some of the database servers are shared, we cannot provide the report directly to the client, as it may contain data from other clients.

**Reply by zubkov-andrei** [reply]
Hm, it seems such use case will cause changes in collection procedures. Right now, pg_profile collects only top-n queries of the whole cluster. If you need to provide workload data for clients on per-database level, we should collect top-n queries for each database rather than for whole cluster. The same applies to schema object statistics. Every sample taken with such approach will be much bigger in size, and might take more resources to process.
It seems we can just hide clusterwide statictics in client report to avoid misunderstanding. Those should be important to the DBA of a whole cluster, anyway.
So, it seems possible, but I can't promise that I'll do it soon.

**Reply by WorstPIlot** [reply]
In my particular case, it is enough to collect all the information as usual, just (possibly) exclude all databases from the final report except the requested one, perhaps it will be possible to implement it simply as a separate function without making fundamental changes?

**Reply by zubkov-andrei** [reply, marked as answer]
Yes, it is possible, but some of your clients won't get workload information if they will have heavy neighbors.

**Reply by WorstPIlot** [reply]
This is not a problem in my case.
